spec/requests/api/v2/providers/courses/update_study_mode_spec.rb

```diff
@@ -0,0 +1,108 @@
+require "rails_helper"
+
+describe 'PATCH /providers/:provider_code/courses/:course_code' do
+  let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+
+  def perform_request(updated_study_mode)
+    jsonapi_data = jsonapi_renderer.render(
+      course,
+      class: {
+        Course: API::V2::SerializableCourse
+      }
+    )
+
+    jsonapi_data[:data][:attributes] = updated_study_mode
+
+    patch "/api/v2/providers/#{course.provider.provider_code}" \
+            "/courses/#{course.course_code}",
+          headers: { 'HTTP_AUTHORIZATION' => credentials },
+          params: {
+            _jsonapi: jsonapi_data
+          }
+  end
+  let(:organisation)      { create :organisation }
+  let(:provider)          { create :provider, organisations: [organisation], sites: [site] }
+  let(:user)              { create :user, organisations: [organisation] }
+  let(:payload)           { { email: user.email } }
+  let(:token)             { build_jwt :apiv2, payload: payload }
+
+  let(:course)            {
+    create :course,
+           provider: provider,
+           study_mode: study_mode,
+           site_statuses: [site_status1, site_status2, site_status3, site_status4]
+  }
+  let(:site_status1) { build(:site_status, :part_time_vacancies, site: site) }
+  let(:site_status2) { build(:site_status, :full_time_vacancies, site: site) }
+  let(:site_status3) { build(:site_status, :both_full_time_and_part_time_vacancies, site: site) }
+  let(:site_status4) { build(:site_status, :with_no_vacancies, site: site) }
+  let(:site) { build(:site) }
+  let(:study_mode) { :full_time_or_part_time }
+
+  let(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  let(:permitted_params) do
+    %i[study_mode]
+  end
+
+  before do
+    perform_request(updated_study_mode)
+  end
+
+  context "course has is updated to full time" do
```

Comment:
This probably needs to be reworded `"course has is updated to full time"`

Reply:
Good stop. Thanks 👍